lib/samly/idp_data.ex

```diff
@@ -200,6 +203,17 @@ defmodule Samly.IdpData do
     %IdpData{idp_data | pre_session_create_pipeline: pipeline}
   end
 
+  @spec set_custom_recipient_url(%IdpData{}, map()) :: %IdpData{}
+  defp set_custom_recipient_url(%IdpData{} = idp_data, %{} = opts_map) do
+    consume_url =
+      case Map.get(opts_map, :custom_recipient_url, nil) do
+        nil -> nil
+        url -> String.to_charlist(url)
+      end
```

Comment:
Style nitpick, as `nil` is the default anyhow:

```suggestion
    consume_url =
      case Map.get(opts_map, :custom_recipient_url) do
        nil -> nil
        url -> String.to_charlist(url)
      end
```